### PR TITLE
feat: add lengthEq and lengthNotEq

### LINKED ISSUE #444: RA.lengthGt, RA.lengthLt, RA.lengthGte, RA.lengthLte, RA.lengthEq, RA.lengthNotEq
```
const isLengthGt = curry((vLength, value) => compose(flip(gt)(vLength), length)(value))
const isLengthLt = curry((vLength, value) => compose(flip(lt)(vLength), length)(value))
const isLengthGte = complement(isLengthLt)
const isLengthLte = complement(isLengthGt)
const isLengthEq = curry((vLength, value) => compose(equals(vLength), length)(value))
const isNotLengthEq = complement(isLengthEq)
```

### PR DESCRIPTION
**Pull request by Undistraction**
Adds remaining functions to test of list or string length is equal or notEqual to the supplied length.

Note: I'm unsure of the correct behaviour for `lengthNotEq` when the value supplied does not have a length property. I am returning false, but should we do something else?

Closes #444

### CODE REVIEW

**Comment by char0n** [comment]
CI is complaining about some jsdoc problems.

> Note: I'm unsure of the correct behaviour for lengthNotEq when the value supplied does not have a length property. I am returning false, but should we do something else?

IMHO is absolutely ok to return false in cases like these.

**Comment by Undistraction** [comment]
> CI is complaining about some jsdoc problems.

It's strange - `npm run lint` doesn't pick up those for me locally. Anyway they are now fixed. 

> IMHO is absolutely ok to return false in cases like these.

OK.

File: src/lengthEq.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @func lengthEq
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param {number} valueLength The length of the list or string
+ * @param {number} value The list or string
+ * @return {boolean}
+ * @see {@link http://ramdajs.com/docs/#equals|equals},
```
Comment (char0n) [inline src/lengthEq.js]: @see is on multiple lines, should be on one line, or prettier is doing this?

File: src/lengthEq.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Returns `true` if the supplied list or string has a length equal to `valueLength`.
+ *
+ * @func lengthEq
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param {number} valueLength The length of the list or string
+ * @param {number} value The list or string
```
Comment (char0n) [inline src/lengthEq.js]: `@param {Array|string} value ...`

This also apply to all the `length*` functions that you did before. Include changes to these function into this PR.

File: src/lengthNotEq.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @func lengthNotEq
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param {number} valueLength The length of the list or string
+ * @param {number} value The list or string
+ * @return {boolean}
+ * @see {@link http://ramdajs.com/docs/#equals|equals},
```
Comment (char0n) [inline src/lengthNotEq.js]: Again on multiple lines.

**Comment by Undistraction** [comment]
@char0n Btw don't feel you have to review this if you want to get a release out today. It can wait for the next release - just wanted to close out the remaining functions in that issue and thought it made sense for them all to go in the same release. But I don't think anyone will lose sleep if they don't.

File: src/lengthNotEq.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Returns `true` if the supplied list or string has a length not equal to `valueLength`.
+ *
+ * @func lengthNotEq
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param {number} valueLength The length of the list or string
+ * @param {number} value The list or string
```
Comment (char0n) [inline src/lengthNotEq.js]: `@param {Array|string} value ...`

This also apply to all the `length*` functions that you did before. Include changes to these function into this PR.

File: test/lengthEq.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import * as RA from '../src/index';
+import eq from './shared/eq';
+
+describe('lengthEq', function() {
+  it(`should return true if the length of a list is equal to the supplied length`, function() {
```
Comment (char0n) [inline test/lengthEq.js]:
```js
context('when the length of a list is equal to the supplied length', function() {
  specify('should return true', function() {
    ...
  });
});

File: test/lengthEq.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    eq(RA.lengthEq(3, [1, 2, 3]), true);
+    eq(RA.lengthEq(3, [1, 2, 3, 4]), false);
+    eq(RA.lengthEq(3, [1, 2]), false);
+    eq(RA.lengthEq(0, []), true);
+  });
+
+  it(`should return true if the length of a string is equal to the supplied length`, function() {
```
Comment (char0n) [inline test/lengthEq.js]:
```js
context('the length of a string is equal to the supplied length', function() {
  specify('should return true', function() {
    ...
  });
});

File: test/lengthEq.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    eq(RA.lengthEq(3, 'abc'), true);
+    eq(RA.lengthEq(3, 'abcd'), false);
+    eq(RA.lengthEq(3, 'ab'), false);
+    eq(RA.lengthEq(0, ''), true);
+  });
+
+  it(`should return false for values without a length property`, function() {
```
Comment (char0n) [inline test/lengthEq.js]:
```js
context('when there are values without a length property', function() {
  specify('should return false', function() {
    ...
  });
});

Comment (char0n) [reply test/lengthEq.js]: Do the same for `lensNotEq` tests.

**Comment by char0n** [comment]
@Undistraction to late, I already did it ;] It looks nice, we need just need to do some improvements here. I need `length*` function next week (have a heavy usecases for them) so I want to release today or tomorrow. I think we can get this last PR in easily.

**Comment by Undistraction** [comment]
I've also added @see refs to each of the other `length…` functions to each length function.

**Comment by Undistraction** [comment]
Ah sorry. I use literals for everything as it means I can interpolate into stings without changing the deliverers which is quicker and better for seeing changes in diffs so I'm doing it out of habit. Obviously I'll try and remember to use normal deliverers, but there should be a way of longing for it.

**Comment by char0n** [comment]
@Undistraction https://github.com/char0n/ramda-adjunct/issues/553

This PR can go inside. We'll deal with the string literals in #553.

**Pull request merged**